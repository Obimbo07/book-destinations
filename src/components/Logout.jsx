import { useNavigate } from "react-router-dom";
import { useDispatch } from "react-redux";
import { logoutUser } from "../redux/authentication/userSlice";
import "../stylesheets/logout.css";

const Logout = () => {
  const dispatch = useDispatch();
  const navigate = useNavigate();
  const user = JSON.parse(localStorage.getItem("user"));
  const token = user.authtoken;
  console.log(token);
  const handleLogout = () => {
    dispatch(logoutUser(token)).then(() => {
      navigate("/");
      window.location.reload();
    });
  };

  return (
    <div className="logout-bg">
      <button className="logout" type="button" onClick={handleLogout}>
        Press again to Logout
      </button>
    </div>
  );
};

export default Logout;
