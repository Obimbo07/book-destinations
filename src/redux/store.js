import { configureStore } from "@reduxjs/toolkit";
import destinationsSlice from "./destinations/destinationsSlice";
import userSlice from "./authentication/userSlice";

const loadStateFromLocalStorage = () => {
  try {
    const serializedState = localStorage.getItem("reduxState");
    if (serializedState === null) {
      return undefined;
    }
    return JSON.parse(serializedState);
  } catch (err) {
    return undefined;
  }
};

const loadUserFromLocalStorage = () => {
  try {
    const serializedUser = localStorage.getItem("user");
    if (serializedUser === null) {
      return undefined;
    }
    return JSON.parse(serializedUser);
  } catch (err) {
    return undefined;
  }
};

const store = configureStore({
  reducer: {
    destinations: destinationsSlice,
    current_user: userSlice,
  },
  preloadedState: {
    current_user: loadUserFromLocalStorage(),
    destinations: loadStateFromLocalStorage(),
  },
});

export default store;
