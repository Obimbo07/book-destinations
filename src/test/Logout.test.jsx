import { render } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import { Provider } from "react-redux";
import { it, expect } from "vitest";
import store from "../redux/store";
import Logout from "../components/Logout";

it("Should render Logout component correctly", () => {
  const { container } = render(
    <Provider store={store}>
      <BrowserRouter>
        <Logout />
      </BrowserRouter>
    </Provider>
  );
  expect(container).toMatchSnapshot();
});
