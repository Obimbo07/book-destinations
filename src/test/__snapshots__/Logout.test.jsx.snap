// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Should render Logout component correctly 1`] = `
<div>
  <div
    class="logout-bg"
  >
    <button
      class="logout"
      type="button"
    >
      Press again to Logout
    </button>
  </div>
</div>
`;
